arbitrum-docs/arbos/l1-to-l2-messaging.mdx

```diff
@@ -1,64 +1,86 @@
 
-A retryable is submitted via the `Inbox`'s `createRetryableTicket` method. In the common case, a Retryable's submission is followed by an attempt to execute the transaction (i.e., an _"auto-redeem"_). If the attempt fails or isn't scheduled after the Retryable is submitted, anyone can try to _redeem_ it, by calling the [`redeem`](./precompiles#ArbRetryableTx) method of the [`ArbRetryableTx`](./precompiles#ArbRetryableTx) precompile. The party requesting the redeem provides the gas that will be used to execute the Retryable. If execution of the Retryable succeeds, the Retryable is deleted. If execution fails, the Retryable continues to exist and further attempts can be made to redeem it. If a fixed period (currently one week) elapses without a successful redeem, the Retryable expires and will be [automatically discarded][discard_link], unless some party has paid a fee to [_renew_][renew_link] the Retryable for another full period. A Retryable can live indefinitely as long as it is renewed each time before it expires.
+## Retryable Tickets Lifecycle
 
+Here we walk through the different stages of the lifecycle of a retryable ticket, shown in the diagram below.
+
+<RetryablesLifeCycleDiagram />
+
+1. Creating a retryable ticket is initiated with a transaction call to the `createRetryableTicket` function of the `inbox` contract. This function has a set of parameters which lead to a succesfull creation (and redemption) of the ticket if set carefully:
+
+    - `address to`: The destination L2 contract address
+    - `uint256 l2CallValue`: The callvalue for retryable L2 message
+    - `uint256 maxSubmissionCost`: The maximum amount of gas to be paid for submitting the ticket. This amount is deducted from sender's L2 balance and is directly proportional to the size of the retryable’s data  
+    - `address excessFeeRefundAddress`: The L2 address to which the excess fee is creadited (gasLimit * maxFeePerGas - ticket execution cost)
+    - `address callValueRefundAddress`: The L2 address to which the l2Callvalue if the ticket times out or gets cancelled
+    - `uint256 gasLimit`: Maximum amount of  gas deducted from sender's L2 balance to cover L2 execution of the ticket
+    - `uint256 maxFeePerGas`: The Price bid for L2 execution of the ticket
+    - `bytes calldata data`: ABI encoded data of L2 message
+
+2. As shown in the diagram above, the sender's deposit must be enough to make the submission succeed (i.e, cover the L1 submission and L2 execution of the ticket).
```

Comment:
"enough to make the submission succeed"
It should make the submission AND execution succeed. Should we use another word there? (Same in the diagram)

Reply:
it's more like "it must be enough to make the submission exceed and for the execution to be _attempted_", i.e., for the later, it has to cover the execution cost as per the provided arguements (but that doesn't guarantee success)

Reply:
might be worth explaining the rationale here (maybe at the beginning of the Lifecycle section?) to clarify; i.e., the L1 contract forces you to provide a "reasonable" amount of funds (at least enough to cover x, y, and z), but that doesn't mean we guarantee successful auto-redemption

Reply:
Makes sense. Thanks guys! Will figure out an edit